### ISSUE TITLE
Invalid database-first mapping for MySQL DateTime(6)

### ISSUE THREAD

**Issue by mludlum** [PomeloFoundation/Pomelo.EntityFrameworkCore.MySql#1733]
Please see https://github.com/ErikEJ/EFCorePowerTools/issues/1605.

When using EF Core Power tools, a datetime field with fractional seconds X results in a DateTime field with a MaxLength(X).  This causes an InvalidCastException when EF expects the field to be a string.

**Comment by mguinness** [comment]
Assuming that you're using version 6.0.2 of Pomelo then the [dotnet scaffold](https://learn.microsoft.com/en-us/ef/core/cli/dotnet#dotnet-ef-dbcontext-scaffold) tool would produce the following output:

**DbContext**
```C#
entity.Property(e => e.EndDate).HasMaxLength(6);
entity.Property(e => e.StartDate).HasMaxLength(6);
```

**Class**
```C#
public DateTime StartDate { get; set; }
public DateTime? EndDate { get; set; }
```

I suspect that EF Core Power Tools should not be adding the `MaxLength` attribute for `DateTime` fields in the class definition.

**Comment by mludlum** [comment]
I am using 6.02.  I checked my generated dbcontext and there were no HasMaxLength calls for any of my DateTime fields.  
Why would HasMaxLength being generated?  Does that apply to DateTime properties with fractional seconds?  This is the documentation for HasMaxLength which states they are only set on array properties.

        /// <summary>
        ///     Configures the maximum length of data that can be stored in this property.
        ///     Maximum length can only be set on array properties (including <see cref="string" /> properties).
        /// </summary>

**Comment by mguinness** [comment]
Just to clarify that `InvalidCastException` is getting raised by the MVC model validation and not within EF Core.

I assume the reason your `DbContext` doesn't include `HasMaxLength` is that you checked "Use DataAnnotation attributes to configure the model" in EF Core Power Tools.  Try again with it unchecked and see what is generated.

**Comment by mludlum** [comment]
Yes, that gets rid of all of my attributes, including ones that I need for model validation.  Still confused why HasMaxLength is added in the DbContext when the method has nothing to do with Dates.

**Comment by mguinness** [comment]
I think you're right that it shouldn't be including precision for that type.  The scaffolding code pulls in column precision or width directly from the MySQL schema information table.

https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/blob/a896e1f2dba5718a42060e78dcc23b6e8f349036/src/EFCore.MySql/Storage/Internal/MySqlTypeMappingSource.cs#L356-L361

Then the code generation uses that information within data type mapping.

https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/blob/a896e1f2dba5718a42060e78dcc23b6e8f349036/src/EFCore.MySql/Scaffolding/Internal/MySqlDatabaseModelFactory.cs#L484-L508

It might be possible to intercept and modify that mapping in [MySqlCodeGenerationMemberAccessTypeMapping](https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/blob/master/src/EFCore.MySql/Scaffolding/Internal/MySqlCodeGenerationMemberAccessTypeMapping.cs).
